src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -391,23 +390,39 @@ internal static unsafe void FixupMethodCell(IntPtr hModule, MethodFixupCell* pCe
         }
 
 #if TARGET_WINDOWS
-        private static unsafe IntPtr GetProcAddressWithSuffix(IntPtr hModule, byte* methodName, byte suffix)
+        private static unsafe IntPtr GetProcAddressWithMangling(IntPtr hModule, byte* methodName, MethodFixupCell* pCell)
+        {
+            IntPtr pMethod = Interop.Kernel32.GetProcAddress(hModule, methodName);
+#if TARGET_X86
+            if (pMethod == IntPtr.Zero && pCell->IsStdcall)
+            {
+                int nameLength = string.strlen(methodName);
+                // We need to add an extra bytes for the prefix, null terminator and stack size suffix ('@' and up to 10 digits)
+                byte* probedMethodName = stackalloc byte[nameLength + 13];
+                probedMethodName[0] = (byte)'_';
+                Unsafe.CopyBlock(probedMethodName + 1, methodName, (uint)nameLength);
+                probedMethodName[nameLength + 1] = (byte)'@';
+                pCell->SignatureBytes.TryFormat(new Span<byte>(probedMethodName + 2 + nameLength, 10), out int bytesWritten);
```

Comment:
It may be nice to spanify this so that a bug in the max buffer size estimation does not lead to buffer overrun. Also, it may be nice to use array pool for large buffers (unbounded stackallocs tend to be flagged by static analyzers).

```
ReadOnlySpan<byte> name = MemoryMarshal.CreateReadOnlySpanFromNullTerminated(methodName);

int bytesNeeded = name.Length + ...;

byte[]? rentedBytes = null;
Span<byte> mangledName = (bytesNeeded <= 128) ? stackalloc byte[128] : (rentedBytes = ArrayPool<byte>.Shared.Rent(bytesNeeded));

mangledName[0] = (byte)'_';
name.CopyTo(mangledName.Slice(1));
mangledName[name.Length + 1] = (byte)'_';
pCell->SignatureBytes.TryFormat(mangledName.Slice(name.Length + 2), out int bytesWritten);
mangledName[name.Length + 2 + bytesWritten] = 0;
```

Reply:
While I agree it would be reasonable thing to do I modeled it after the existing code that handles the `A`/`W` suffixes. Unless you feel really strongly about it I would prefer to do this clean up in a separate PR.

Reply:
Yes, that's fine. I do not feel strongly about it.